packages/trpc/server/routers/viewer/bookings/reportBooking.handler.ts

```diff
@@ -0,0 +1,186 @@
+      (await isTeamOwner(user.id, booking.eventType.teamId));
+  }
+
+  // For team admins, also check if they can see this booking through team membership
+  // This matches the logic in get.handler.ts where team admins can see bookings from their team members
+  let canAccessThroughTeamMembership = false;
+  if (isTeamAdminOrOwner) {
+    // If user is team admin/owner, they can report any booking that appears in their booking list
+    // This includes bookings where team members are attendees
+    canAccessThroughTeamMembership = true;
+  }
+
+  if (!isBookingOwner && !isAttendee && !canAccessThroughTeamMembership) {
+    throw new TRPCError({ code: "FORBIDDEN", message: "You don't have access to this booking" });
+  }
+
+  // Check if user has already reported this booking or any in the recurring series
+  const existingReports = booking.reports;
+
+  if (booking.recurringEventId) {
+    // For recurring bookings, check if user has reported any booking in the series
+    const recurringReports = await prisma.bookingReport.findMany({
+      where: {
+        reportedById: user.id,
+        booking: {
+          recurringEventId: booking.recurringEventId,
+        },
+      },
+    });
+
+    if (recurringReports.length > 0) {
+      throw new TRPCError({
+        code: "BAD_REQUEST",
+        message: "You have already reported this recurring booking series",
+      });
+    }
+  } else if (existingReports.length > 0) {
+    throw new TRPCError({ code: "BAD_REQUEST", message: "You have already reported this booking" });
+  }
+
+  // For recurring bookings, report all remaining instances
+  let reportedBookingIds: number[] = [bookingId];
+
+  if (booking.recurringEventId) {
+    // Get all future bookings in the recurring series
+    const futureRecurringBookings = await prisma.booking.findMany({
+      where: {
+        recurringEventId: booking.recurringEventId,
+        startTime: {
+          gte: new Date(),
+        },
+      },
+      select: {
+        id: true,
+      },
+    });
+
+    reportedBookingIds = futureRecurringBookings.map((b) => b.id);
+  }
+
+  // Create reports for all relevant bookings
+  await prisma.bookingReport.createMany({
+    data: reportedBookingIds.map((id) => ({
+      bookingId: id,
+      reportedById: user.id,
+      reason,
+      description,
+      cancelled: cancelBooking,
+    })),
+  });
+
+  const bookingReport = { id: reportedBookingIds[0] }; // For backward compatibility
+
+  // Cancel booking if requested and conditions are met
+  let cancellationError = null;
+  if (
+    cancelBooking &&
+    (booking.status === BookingStatus.ACCEPTED || booking.status === BookingStatus.PENDING) &&
+    new Date(booking.startTime) > new Date()
+  ) {
+    try {
+      await handleCancelBooking({
+        bookingData: {
+          uid: booking.uid,
+          cancelledBy: user.email,
+          cancellationReason: "Booking cancelled due to report",
```

Comment:
As mentioned before, the attendee shouldn't be informed that the booking was reported. We don't need a cancellation reason